src/Components/ProfilePage.js

```diff
@@ -1,41 +1,79 @@
-            program: "SYDE - 3A",
-            socialMedia: "@LuluLiu",
-            avatar: Images.avatar.chicken
-            }
+            user : this.props.userInfo,
+            token : this.props.userToken
+        }
+        this.handleLogout = this.handleLogout.bind(this);
+        this.handleProfileUpdate = this.handleProfileUpdate.bind(this);
     }
+
+    handleProfileUpdate = (content)  => {
+        this.setState({user : content});
+        const {firstName, lastName, programName, location, term, socialLink, avatar} = this.state.user;
+        axios.put(`http://localhost:8080/users`, {
+             firstName : firstName,
+             lastName : lastName,
+             programName : programName,
+             location : location,
+             term : term,
+             socialLink : socialLink,
+             avatar : avatar,
+             token: this.state.token
+            })
+          .then(res => {
+              console.log(res.status);
+              if(res.status == 200) {
+                console.log("Update Successful");
+              }
+          })
+    }
+
+    handleLogout(){
+        return new Promise((resolve, reject) => {
+            axios.post(`http://localhost:8080/logout`, {
+                    token: this.state.token
+                })
+            .then(res => {
+                if(res.status == 200) {
+                    console.log("Logout Successful");
+                    resolve({ redirectPath: "/", params: {}});
+                }
+            })
+        })
+    }
+
     render() {
+
         return(
             <div className="app-page" id="profile-page">
                 <div className="app-page-fill"></div>
                     <MDBRow>
-                        <img className="avatar" src ={this.state.avatar}></img>
+                        <img className="avatar" src ={this.state.user.avatar? Images.avatar[this.state.user.avatar.toLowerCase()] : Images.avatar.goose}></img>
                     </MDBRow>
                 <div className="app-main-section">
                     <div className="page-body">
                         <MDBRow>
                                 <div className="description">
-                                    <h2><b>{this.state.name}</b></h2>
-                                    <h4>{this.state.program}</h4>
-                                    <h4>{this.state.socialMedia}</h4>
-                                    <p>{this.state.biography}</p>
+                                    <h2><b>{this.state.user.firstName} {this.state.user.lastName ? this.state.user.lastName[0]:this.state.user.lastName}.</b></h2>
```

Comment:
Curious with the ternary? When would last name be an array?

Reply:
We want to show last initial only.
I set it as ternary because in prod there are times right now that we don't have data loaded in and if we request an element of an array that doesn't exist, profile page would flip shit

Reply:
Can we make the default value (the thing after the colon) an empty string then? Just so it's clearer to someone seeing that we'll have nothing if there's no last name to take the first letter. UX makes sense though!